client/src/components/expectations-list.tsx

```diff
@@ -32,26 +33,69 @@ import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
 import HintsList from "components/hints-list";
 import { LessonExpectation, Hint } from "types";
 import "styles/layout.css";
+import "jsoneditor-react/es/editor.min.css";
 
 const ExpectationCard = (props: {
   classes: any;
+  loaded: boolean;
   expectation: LessonExpectation;
   expIdx: number;
   canDelete: boolean;
   handleExpectationChange: (val: string) => void;
   handleRemoveExpectation: () => void;
   handleHintChange: (val: Hint[]) => void;
+  handleFeaturesChange: (val: any) => void;
 }) => {
   const {
     classes,
+    loaded,
     expectation,
     expIdx,
     canDelete,
     handleExpectationChange,
     handleRemoveExpectation,
     handleHintChange,
+    handleFeaturesChange,
   } = props;
   const [expanded, setExpanded] = React.useState(true);
+  const editorRef = React.useRef<any>();
+
+  const ajv = new Ajv({ allErrors: true, verbose: true });
+  /* eslint-disable-next-line @typescript-eslint/no-var-requires */
+  const schema = require("schemas/expectation-feature-schema.json");
```

Comment:
can move this up to module level (immediately after imports)? Otherwise, I think may be needlessly reading in the file on every render?